### PR TITLE
[DADZ-355] Enable routing to accept tokens to be excluded when finding a suitable route

### PR DESCRIPTION
**Pull request by 0xpApaSmURf**
- feature

Currently all pools are considered when finding routes to tokens, however to support restricted tokens we should be able to specify tokens that should be avoided when finding routes

This PR allows passing a list of token addresses that must be excluded from the route search and return only routes that do not include such tokens.

### CODE REVIEW

File: src/routers/alpha-router/functions/get-candidate-pools.ts [hunk 1]
```diff
@@ -569,9 +584,26 @@ export async function getV3CandidatePools({
...
+  // Only consider pairs where neither tokens are to be excluded.
+  let filteredTokenPairs: [Token, Token, FeeAmount][] = tokenPairs;
+  if (excludeTokens && excludeTokens.length > 0) {
+    filteredTokenPairs = [];
+    for (const [token0, token1, fee] of tokenPairs) {
+      const excludeToken0 = excludeTokens.find((ex) => token0.address == ex);
+      const excludeToken1 = excludeTokens.find((ex) => token1.address == ex);
+
+      console.log(excludeToken0);
+      console.log(excludeToken1);
```
Comment (sudoFerraz) [inline src/routers/alpha-router/functions/get-candidate-pools.ts:596]: do we want to keep this here?

File: test/unit/routers/alpha-router/functions/get-candidate-pools.test.ts [hunk 2]
```diff
@@ -170,6 +170,34 @@ describe('get candidate pools', () => {
     ).toBeTruthy();
   });
 
+  test('succeeds to get top pools while excluding WETH', async () => {
```
Comment (sudoFerraz) [inline test/unit/routers/alpha-router/functions/get-candidate-pools.test.ts:173]: Should we also add a test for the scenario where WETH would not be excluded (asking route as Acc_inv meaning no tokens need to be excluded)

**Review by sudoFerraz — APPROVED** [review]

File: src/routers/alpha-router/functions/get-candidate-pools.ts (hunk 1 above)
Comment (0xpApaSmURf) [reply src/routers/alpha-router/functions/get-candidate-pools.ts]: Already removed ;)

File: test/unit/routers/alpha-router/functions/get-candidate-pools.test.ts (hunk 2 above)
Comment (0xpApaSmURf) [reply test/unit/routers/alpha-router/functions/get-candidate-pools.test.ts:173]: The other tests in the file do exactly that :D

**Pull request merged**